### PR TITLE
Feature/fga/voip switch call and transfer init

### PR DESCRIPTION
**Pull request by ganfra**
This  PR introduces:
- Call switching, so you can have 1 active call and 1 on hold (or 2 on hold) (#2533)
- Call transfer init (reusing user list fragment) (#2553)
- Elapsed time on call

It also fixes a bunch of issues.

**Switch call UI:** 

**Call transfer UI:**

### CODE REVIEW

**Review by bmarty — CHANGES_REQUESTED** [review]
Nice work. A few remarks and questions about the implementation.

File: vector/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -2773,4 +2774,18 @@
     <string name="call_tile_other_declined">%1$s declined this call</string>
     <string name="call_tile_ended">This call has ended</string>
     <string name="call_tile_call_back">Call back</string>
+
+
+    <string name="call_only_active">Active call (%1$s)</string>
+    <string name="call_only_paused">Paused call</string>
+    <string name="call_active_and_single_paused">1 active call (%1$s) · 1 paused call</string>
+    <string name="call_active_and_multiple_paused">1 active call (%1$s) · %2$d paused calls</string>
+    <string name="call_only_multiple_paused">%1$d paused calls</string>
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/res/values/strings.xml]: done

File: vector/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -2773,4 +2774,18 @@
     <string name="call_tile_other_declined">%1$s declined this call</string>
     <string name="call_tile_ended">This call has ended</string>
     <string name="call_tile_call_back">Call back</string>
+
+
+    <string name="call_only_active">Active call (%1$s)</string>
+    <string name="call_only_paused">Paused call</string>
+    <string name="call_active_and_single_paused">1 active call (%1$s) · 1 paused call</string>
+    <string name="call_active_and_multiple_paused">1 active call (%1$s) · %2$d paused calls</string>
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/res/values/strings.xml]: done

File: vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt [hunk 3]
```diff
@@ -68,21 +68,15 @@ class UserListViewModel @AssistedInject constructor(@Assisted initialState: User
     }
 
     init {
-        setState {
-            copy(
-                    myUserId = session.myUserId,
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/userdirectory/UserListViewModel.kt:73]: It's not used at all, I forgot to remove it entirely from the state

File: vector/src/main/java/im/vector/app/features/invite/InviteUsersToRoomActivity.kt [hunk 4]
```diff
@@ -92,15 +95,14 @@ class InviteUsersToRoomActivity : SimpleFragmentActivity(), UserListViewModel.Fa
...
-            val args: InviteUsersToRoomArgs? = intent.extras?.getParcelable(MvRx.KEY_ARG)
             addFragment(
                     R.id.container,
                     UserListFragment::class.java,
                     UserListFragmentArgs(
                             title = getString(R.string.invite_users_to_room_title),
                             menuResId = R.menu.vector_invite_users_to_room,
                             excludedUserIds = viewModel.getUserIdsOfRoomMembers(),
-                            existingRoomId = args?.roomId
+                            showInviteActions = false
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/invite/InviteUsersToRoomActivity.kt:105]: Yes, it was just used to know if we show invite actions. So replaced it directly by the bool

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/RoomSummaryHolder.kt [hunk 5]
```diff
@@ -16,25 +16,28 @@
...
 
 /*
-    This holds an instance of the current room summary.
-    You should use this in the context of the timeline.
+    You can use this to share room summary instances within the app.
+    You should probably use this only in the context of the timeline
  */
-@ScreenScope
+@Singleton
 class RoomSummaryHolder @Inject constructor() {
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/helper/RoomSummaryHolder.kt]: done

File: vector/src/main/java/im/vector/app/features/call/webrtc/WebRtcCallManager.kt [hunk 6]
```diff
@@ -63,11 +66,11 @@ class WebRtcCallManager @Inject constructor(
         get() = activeSessionDataSource.currentValue?.orNull()
 
     interface CurrentCallListener {
-        fun onCurrentCallChange(call: WebRtcCall?)
+        fun onCurrentCallChange(call: WebRtcCall?) {}
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/webrtc/WebRtcCallManager.kt:69]: I think it's ok on for listeners...

File: vector/src/main/java/im/vector/app/features/call/transfer/CallTransferViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+    override fun onCleared() {
+        super.onCleared()
+        call?.removeListener(callListener)
+    }
+
+    override fun handle(action: CallTransferAction) {
+        when (action) {
+            is CallTransferAction.Connect -> transferCall(action)
+        }
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/transfer/CallTransferViewModel.kt]: Done

File: vector/src/main/java/im/vector/app/features/call/transfer/CallTransferViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    private val callListener = object : WebRtcCall.Listener {
+        override fun onStateUpdate(call: MxCall) {
+            if (call.state == CallState.Terminated) {
+                _viewEvents.post(CallTransferViewEvents.Dismiss)
+            }
+        }
+    }
+
+    init {
+        val webRtcCall = callManager.getCallById(initialState.callId)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/features/call/transfer/CallTransferViewModel.kt]: Done

File: vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt [hunk 9]
```diff
@@ -146,6 +147,7 @@ interface ScreenComponent {
     fun inject(activity: VectorJitsiActivity)
     fun inject(activity: SearchActivity)
     fun inject(activity: UserCodeActivity)
+    fun inject(callTransferActivity: CallTransferActivity)
```
[the remark this answers is not in the document]
Comment (ganfra) [reply vector/src/main/java/im/vector/app/core/di/ScreenComponent.kt]: Done

File: vector/src/main/java/im/vector/app/core/di/VectorComponent.kt [hunk 10]
```diff
@@ -159,7 +159,7 @@ interface VectorComponent {
 
     fun webRtcCallManager(): WebRtcCallManager
 
-    fun roomSummaryHolder(): RoomSummaryHolder
+    fun roomSummaryHolder(): RoomSummariesHolder
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/core/di/VectorComponent.kt:162]: Not a big deal but this should also have to be renamed.

**Review by bmarty — APPROVED** [review]
Thanks for the update, you can merge into your main branch whenever you want to!

**Pull request merged**